sim/battle.ts

```diff
@@ -2185,16 +2185,13 @@ export class Battle extends Dex.ModdedDex {
 			} else {
 				target = pokemon.side.active[-targetLoc - 1];
 			}
-			if (target) {
-				if (!target.fainted) {
-					// target exists and is not fainted
-					return target;
-				} else if (target.side === pokemon.side) {
-					// fainted allied targets don't retarget
```

Comment:
Is this wrong?

Reply:
They don't retarget by default, but still can be redirected by Storm Drain / Lightningrod.

Reply:
BTW, it might be unclear, but this diff is equivalent to just changing the ``return null`` for ``return target``.

Reply:
Is the prevent-retargeting code done elsewhere or something?

Reply:
Oh, I understand now.